File: LeetCode/Explore/TopInterview/Amazon/SortingSearching/973_k_closest_points_to_origin.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        mp = dict()
        
        for i, point in enumerate(points):
          mp[i] = point[0] * point[0] + point[1] * point[1]
          
        vp = list(k for k, v in sorted(mp.items(), key = lambda x:x[1]))[:k]
        
        ans = list()
        
        for v in vp:
          ans.append([points[v][0], points[v][1]])
          
        return ans
          

File: LeetCode/Explore/TopInterview/Amazon/SortingSearching/973_k_closest_points_to_origin.cpp
class Solution {
public:
    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
      typedef pair<int, int> P;
      priority_queue<P, vector<P>, greater<P> > pq; // ascending
      vector<vector<int> > ans;
      for (int i = 0; i < points.size(); i++) {
        pq.push(make_pair(pow(points[i][0], 2) + pow(points[i][1], 2), i));
      }
      while (k) {
        ans.push_back(points[pq.top().second]);
        pq.pop();
        k--;
      }
      return ans;
    }
};
